File: src/CHALLENGE_01/solucion.py
def resolve(input_string: str) -> str:
    words = input_string.split(" ")
    normalized_words = [word.lower() for word in words]

    count = {}
    for word in normalized_words:
        if word not in count.keys():
            count[word] = 0

        count[word] += 1

    return "".join([f"{word}{count[word]}" for word in count.keys()])

File: src/CHALLENGE_01/solucion.ts
function resolve(input_string: string): string {
    const wordCount = input_string
        .split(" ")
        .map((word) => word.toLowerCase())
        .reduce<Record<string, number>>((count, word) => {
            if (word in count) {
                count[word] += 1;
                return count;
            }

            count[word] = 1;
            return count;
        }, {});

    const keys = Object.keys(wordCount);

    const wordsWithNumbers = keys.map((key) => {
        return `${key}${wordCount[key]}`;
    });
    
    return wordsWithNumbers.join("");
}
